src/ui/wxWidgets/AddEditPropSheetDlg.cpp

```diff
@@ -2656,16 +2668,40 @@ void AddEditPropSheetDlg::OnExpRadiobuttonSelected( wxCommandEvent& evt )
   bool Never = (evt.GetEventObject() == m_DatesTimesNeverExpireCtrl);
 
   if (Never) {
-    m_ExpirationTime = _("Never");
-    m_CurrentExpirationTime.Clear();
-    m_tttExpirationTime = time_t(0);
-    m_ExpirationTimeInterval = 90;
-    wxDateTime xdt(wxDateTime::Now());
+    //m_tttExpirationTime = time_t(0);
+    // If it has an interval, keep it.  Otherwise, use the user defined default.
+    // This applies to new entries and old entries with no interval setting.
+    //
+    // Some entries, created with recent versions of pwsafe, will have an interval but no date,
+    // which is interpreted as "Never".  But, we need to use the given interval to prevent
+    // a false change detection when the entry is opened and closed.  If an actual change is made,
+    // the entry will be re-written correctly.
+    if (!m_ExpirationTimeInterval) { // Compatibility
```

Comment:
To confirm, there's no "never" setting anymore - if the user hasn't specified any expiration settings, we'll always use the default interval - is that right?
E.g. if the user selected "never", why do we set the expiry date? Wouldn't that cause the entry to be marked expired on that date, contrary to what the user wanted?

And for additional context, is the problem basically that there was a bug which caused a nonsensical combination of settings to get stored (interval but no date), so we're doing a one-time migration to bring the stored settings to a valid state?

Reply:
> To confirm, there's no "never" setting anymore - if the user hasn't specified any expiration settings, we'll always use the default interval - is that right?

There never was a "never" parameter written to the file, it's the absence of the date and interval fields.  Upon initialization of the dialog, my intention is to pre-populate the controls with the default interval and the corresponding date in case the user wants to use them, but the "Never" radio button will be selected.

> E.g. if the user selected "never", why do we set the expiry date? Wouldn't that cause the entry to be marked expired on that date, contrary to what the user wanted?

The control values won't matter if the "never" radio button is selected, neither field is written to the file.  Also, Never is still the default for a new entry.

> And for additional context, is the problem basically that there was a bug which caused a nonsensical combination of settings to get stored (interval but no date)

Exactly, the current version is doing that when Never is selected, so there could be lots of files with that condition.  

> so we're doing a one-time migration to bring the stored settings to a valid state?

No, I didn't want to make mysterious changes to the file.  I accept that condition as Never. (As does the current release.) But, if the user makes an expiry change, it will be written correctly.  One important goal to me is that the user can open the edit dialog and cancel without any changes and not have a false change detection pop-up or write changes to the file.
I did use that interval field to pre-populate the control.  I'm starting to re-consider that, though.  Maybe I should use the default value and completely ignore that field.

Any additional thoughts are welcome.